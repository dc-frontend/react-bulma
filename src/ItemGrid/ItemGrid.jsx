import React from 'react'
import { Column, Columns } from '../Columns/Columns'

export class ItemGrid extends React.Component {
  key () {
    return this.props.key || 'id'
  }

  columns () {
    return this.props.columns || 3
  }

  renderColumns (columns) {
    const cols = columns.map(col => {
      return this.renderColumn(col)
    })

    return (
      <Columns>
        {cols}
      </Columns>
    )
  }

  renderColumn (col) {
    const items = col.items.map(item => {
      return item
    })

    return (
      <Column key={col.id}>
        {items}
      </Column>
    )
  }

  buildColumnizedListing () {
    let items = this.props.children
    let colCount = this.columns()
    let columns = []

    let i = 0
    while (i < colCount) {
      columns.push({ id: i, items: [] })
      i++
    }

    console.log('typeof items', typeof items)
    console.log('items', items)

    if (!Array.isArray(items)) {
      columns[0].items.push(this.props.children)
      return columns
    }

    let column = 0
    items.forEach(item => {
      columns[column].items.push(item)

      if (column === 2) {
        column = 0
      } else {
        column++
      }
    })
    return columns
  }

  title () {
    return <h1>{this.props.title}</h1> || ''
  }

  render () {
    let data = this.buildColumnizedListing()
    let title = this.title()
    let cols = this.renderColumns(data)
    let outerClasses = 'item-grid '
    if (typeof this.props.outerClasses !== 'undefined') {
      outerClasses = outerClasses + this.props.outerClasses
    }

    return (
      <div className={outerClasses}>
        {title}
        {cols}
      </div>
    )
  }
}

export default ItemGrid
