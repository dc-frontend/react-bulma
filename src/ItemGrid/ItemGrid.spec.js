// Link.react.test.js
import React from 'react'
import ItemGrid from './ItemGrid'
import renderer from 'react-test-renderer'
import { cleanup } from '@testing-library/react'

afterEach(cleanup)

test('It renders as expected with no content', () => {
  const component = renderer.create(
    <ItemGrid></ItemGrid>
  )
  expect(component.toJSON()).toMatchSnapshot()
})
