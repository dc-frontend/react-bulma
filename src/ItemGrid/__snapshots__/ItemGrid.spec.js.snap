// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders as expected with no content 1`] = `
<div
  className="item-grid "
>
  <h1 />
  <div
    className="columns"
  >
    <div
      className="column"
    />
    <div
      className="column"
    />
    <div
      className="column"
    />
  </div>
</div>
`;
